Validation jfc

Jacob-F-Cann opened the pull request:
Push initial_validation_jfc.ipynb so Ojasv view and edit my additions

kessler24 commented:
My GitHub is indicating that there are branch conflicts between your branch and main. You will need to pull main into your branch and resolve the conflicts before the PR can be approved.

Jacob-F-Cann commented:
> I pulled your code and tried to run it in the docker environment, but there are several errors. I've created a commit that should fix them and pushed to this PR.
> 
> First, I was getting an error on this line `yearly = df['Anomaly'].resample('YE').mean().reset_index()` (in code section titled 1. Convert daily data → yearly averages) that says `ValueError: Invalid frequency: YE`. I fixed it by changing the frequency to 'A', which is the older version of YE. It seems likely there is something weird going on with the different interactions in the packages that is causing pandas to not recognize YE, but I haven't fully traced the issue.
> 
> Second, in this line during data validation: `try: df = target_train_schema.validate(train_df)` df gets overwritten with train_df, such that when you get to machine learning there is an error because test_df is empty. I fixed it by setting `train_df =` instead.

1. I saw this too not sure why. Maybe YE is supposed to be Year?
2. I think we need to use deep copy so df modifications do propagate accidentally between cells.

Jacob-F-Cann commented:
> Hey Jacob. It looks like you may have accidentally deleted the github workflow that creates the docker environment (github/workflows/docker-publish.yml file was deleted). Can you please undo that change. It also looks like you published a new docker image through the automatic workflow (maybe before it was deleted)? Could you clarify if that was intentional (and if so, what you changed) or an accidental trigger?

Yes it was unintentional, I don't know what I did but I was trying not to push unnecessary files, idk why I gitignored the yml that was mistake.
